third_party/jpeg/jpeg.BUILD

```diff
@@ -274,50 +276,118 @@ cc_library(
+    out = "jconfigint_nowin.h",
+    substitutions = {
+        "#undef BUILD" : "#define BUILD \"20161115\"",
+        "#undef inline" : "",
+        "#undef INLINE" : "#define INLINE inline __attribute__((always_inline))",
+        "#undef PACKAGE_NAME" : "#define PACKAGE_NAME \"libjpeg-turbo\"",
+        "#undef VERSION" : "#define VERSION \"1.5.1\"",
+        "#undef SIZEOF_SIZE_T" : "#if (__WORDSIZE==64 && !defined(__native_client__))\n" +
+            "#define SIZEOF_SIZE_T 8\n" +
+            "#else\n" +
+            "#define SIZEOF_SIZE_T 4\n" +
+            "#endif\n",
+    },
+    template = "jconfigint.h.in",
+)
+
 genrule(
     name = "configure",
     outs = ["jconfig.h"],
-    cmd = "cat <<'EOF' >$@\n" +
-          "#define JPEG_LIB_VERSION 62\n" +
-          "#define LIBJPEG_TURBO_VERSION 1.5.1\n" +
-          "#define LIBJPEG_TURBO_VERSION_NUMBER 1005001\n" +
-          "#define C_ARITH_CODING_SUPPORTED 1\n" +
-          "#define D_ARITH_CODING_SUPPORTED 1\n" +
-          "#define BITS_IN_JSAMPLE 8\n" +
-          "#define HAVE_LOCALE_H 1\n" +
-          "#define HAVE_STDDEF_H 1\n" +
-          "#define HAVE_STDLIB_H 1\n" +
-          "#define HAVE_UNSIGNED_CHAR 1\n" +
-          "#define HAVE_UNSIGNED_SHORT 1\n" +
-          "#define MEM_SRCDST_SUPPORTED 1\n" +
-          "#define NEED_SYS_TYPES_H 1\n" +
-          select({
-              ":k8": "#define WITH_SIMD 1\n",
-              ":armeabi-v7a": "#define WITH_SIMD 1\n",
-              ":arm64-v8a": "#define WITH_SIMD 1\n",
-              "//conditions:default": "",
-          }) +
-          "EOF",
+    srcs = [
+        ":jconfig_win",
+        ":jconfig_nowin_nosimd",
+        ":jconfig_nowin_simd",
+    ],
+    cmd = select({
+        ":windows": "cp $(locations :jconfig_win) $(@D)/jconfig.h",
+        ":k8": "cp $(locations :jconfig_nowin_simd) $(@D)/jconfig.h",
+        ":armeabi-v7a": "cp $(locations :jconfig_nowin_simd) $(@D)/jconfig.h",
+        ":arm64-v8a": "cp $(locations :jconfig_nowin_simd) $(@D)/jconfig.h",
+        "//conditions:default": "cp $(locations :jconfig_nowin_nosimd) $(@D)/jconfig.h",
+    }),
 )
 
 genrule(
     name = "configure_internal",
     outs = ["jconfigint.h"],
-    cmd = "cat <<'EOF' >$@\n" +
-          "#define BUILD \"20161115\"\n" +
-          "#ifdef _MSC_VER  /* Windows */\n" +
-          "#define INLINE __inline\n" +
-          "#else\n" +
-          "#define INLINE inline __attribute__((always_inline))\n" +
-          "#endif\n" +
-          "#define PACKAGE_NAME \"libjpeg-turbo\"\n" +
-          "#define VERSION \"1.5.1\"\n" +
-          "#if (__WORDSIZE==64 && !defined(__native_client__)) || defined(_WIN64)\n" +
-          "#define SIZEOF_SIZE_T 8\n" +
-          "#else\n" +
-          "#define SIZEOF_SIZE_T 4\n" +
-          "#endif\n" +
-          "EOF",
+    srcs = [
+        ":jconfigint_win",
+        ":jconfigint_nowin",
+    ],
+    cmd = select({
+        ":windows": "cp $(locations :jconfigint_win) $(@D)/jconfigint.h",
```

Comment:
Done.
One more thing, it turns out if I use "jconfigint_nowin.h" in srcs, I have to refer to it the same way here as I do in srcs.